### ISSUE TITLE
consistent-function-scoping false positive for uninitialized let variable

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#586]
Running ESLint 6.8.0 with eslint-plugin-unicorn 17.1.0 and `"extends": "plugin:unicorn/recommended"` on a file with the following reduced example code:

```js
'use strict';

function cat(stream) {
  let content;

  function onEnd() {
    process.stdout.write(content);
  }

  stream
    .setEncoding('utf8')
    .on('data', (chunk) => {
      if (!content) {
        content = chunk;
      } else {
        content += chunk;
      }
    })
    .once('end', onEnd);
}

cat(process.stdin);
```
gives the following output:

    /path/to/example.js
      6:3  error  Move function to the outer scope  unicorn/consistent-function-scoping
    
    ✖ 1 problem (1 error, 0 warnings)

The error seems incorrect since the `onEnd` function can't be moved to the outer scope because it references `content` defined in the inner scope.

Note that if `let content;` is changed to `let content = '';` the error goes away.

Thanks,
Kevin

**Comment by kevinoid** [comment]
Thanks for the very fast fix @fisker and @sindresorhus!  Works great for me.
